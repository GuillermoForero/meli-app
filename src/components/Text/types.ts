export type TextDecoration =
  | 'underline'
  | 'line-through'
  | 'overline';

export type TextAlign =
  | 'left'
  | 'center'
  | 'right';

export interface TextProps {
  color?: string;
  textAlign?: TextAlign;
  fontSize?: string;
  fontWeight?: string;
  margin?: string;
  className?: string;
  txtDecoration?: TextDecoration;
}
