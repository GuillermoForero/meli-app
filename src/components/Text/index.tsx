import styled from 'styled-components';
import { TextProps } from 'components/Text/types';
import { COLORS } from 'consts';
//I created this reusable component because for SEO it would not affect much, due to the behavior of the website itself and the react
const Text = styled.p.attrs((props: TextProps) => ({
  ...props,
  color: props.color || COLORS.GRAY_20,
}))`
  /* DEFAULT DEFINITIONS */
  font-family: 'Montserrat', sans-serif;
  margin: ${({ margin }) => margin? margin: '0'};
  text-decoration: ${({ txtDecoration }) =>
    txtDecoration ? txtDecoration : 'none'};
  font-size: ${({ fontSize }) => fontSize? fontSize: '12px'};
  font-weight: ${({ fontWeight }) => fontWeight? fontWeight: '400'};
  text-align: ${({ textAlign }) => textAlign? textAlign: 'left'};
  color: ${({ color }) => color};
  /* WORD-BREAKING DEFINITIONS */
  white-space: pre-wrap;
  white-space: -moz-pre-wrap;
  white-space: -pre-wrap;
  white-space: -o-pre-wrap;
  word-wrap: break-word;
  -ms-word-break: break-all;
  word-break: break-all;
  /* Non standard for WebKit */
  word-break: break-word;
`;

export default Text;
