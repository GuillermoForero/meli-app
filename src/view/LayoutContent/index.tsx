import Box from 'components/Box';
import Header from 'components/Header';
import { COLORS } from 'consts';
import React from 'react';

const LayoutContent: React.FC = (props) => (
    <Box alignItems="center" flexDirection="column" style={{ backgroundColor: `${COLORS.DEFAULT_GRAY}` }} minHeight="100vh">
    <Header/>
    {props.children}
  </Box>
);

export default LayoutContent;
