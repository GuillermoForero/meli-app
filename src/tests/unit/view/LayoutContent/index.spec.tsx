import React from 'react';
import { render } from 'tests/support/customRender';
import Text from 'components/Text';
import LayoutContent from 'view/LayoutContent';

describe('[LayoutContent]', () => {
  it('Renders default snapshot', () => {
    const { container, getByPlaceholderText } = render(<LayoutContent>text example</LayoutContent>);
    expect(getByPlaceholderText('Nunca dejes de buscar')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
