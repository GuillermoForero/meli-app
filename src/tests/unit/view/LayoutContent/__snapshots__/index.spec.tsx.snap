// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LayoutContent] Renders default snapshot 1`] = `
<div>
  <div
    class="sc-bdfBwQ kkjLqa"
    style="background-color: rgb(238, 238, 238);"
  >
    <div
      class="sc-bdfBwQ dykpMl"
      height="60px"
      style="background-color: rgb(255, 230, 0);"
    >
      <img
        alt="search"
        class="sc-eCssSg kFfXRd"
        src="Logo_ML2x.png"
      />
      <div
        class="sc-bdfBwQ hjxpIr"
        width="auto"
      >
        <input
          class="sc-gsTCUz kebsqW"
          placeholder="Nunca dejes de buscar"
          type="text"
        />
        <button
          class="sc-dlfnbm iNXthm"
          type="submit"
        >
          <img
            alt="search"
            class="sc-hKgILt irWZxG"
            src="ic_Search2x.png"
          />
        </button>
      </div>
    </div>
    text example
  </div>
</div>
`;
